test/e2e/CI/ci_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+package CI
+
+import (
+	"github.com/apache/incubator-yunikorn-k8shim/test/e2e/framework/cfg_manager"
+	"github.com/apache/incubator-yunikorn-k8shim/test/e2e/framework/helpers"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+var _ = ginkgo.Describe("CI: Test for basic scheduling", func() {
+	var kClient helpers.KubeCtl
+	var restClient helpers.RClient
+	var sleepPodDef = cfg_manager.GetAbsPath("../testdata/sleeppod_template.yaml")
+	var sleepRespPod *v1.Pod
+	var dev = "development"
+	var appsInfo map[string]interface{}
+	var r = regexp.MustCompile("memory:(\\d+) vcore:(\\d+)")
+
+	ginkgo.BeforeSuite(func() {
+		// Initializing kubectl client
+		kClient = helpers.KubeCtl{}
+		gomega.Ω(kClient.SetClient()).To(gomega.BeNil())
+		// Initializing rest client
+		restClient = helpers.RClient{}
+		ginkgo.By("create development namespace")
+		ns1, err := kClient.CreateNamespace(dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		gomega.Ω(ns1.Status.Phase).To(gomega.Equal(v1.NamespaceActive))
+
+		ginkgo.By("Deploy the sleep pod to the development namespace")
+		sleepObj, err := helpers.GetPodObj(sleepPodDef)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		sleepObj.Namespace = dev
+		sleepObj.ObjectMeta.Labels["applicationId"] = helpers.GetUUID()
+		sleepRespPod, err = kClient.CreatePod(sleepObj, dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		//Adding sleep here for the app info to be reflected.
+		time.Sleep(3 * time.Second)
```

Comment:
what's the best practice for this?
Usually, we do not want to depend a hard code time to sleep and wait for something to happen, does `gomega` provides functionality to like https://github.com/apache/incubator-yunikorn-k8shim/blob/bdacf71ac449ad54fe8e40ef7086b35e00af5c46/pkg/common/utils/utils.go#L132?